pyvista/core/cell.py

```diff
@@ -1,9 +1,213 @@
+        return self.GetNumberOfEdges()
+
+    @property
+    def point_ids(self) -> List[int]:
+        """Get the point ids building the cell."""
+        point_ids = self.GetPointIds()
+        return [point_ids.GetId(i) for i in range(point_ids.GetNumberOfIds())]
+
+    @property
+    def points(self) -> np.ndarray:
+        """Get the point coordinates of the cell."""
+        # A copy of the points must be returned to avoid overlapping them since the
+        # `vtk.vtkExplicitStructuredGrid.GetCell` is an override method.
+        points = _vtk.vtk_to_numpy(self.GetPoints().GetData())
+        return points.copy()
+
+    @property
+    def edges(self) -> List[Cell]:
+        """Get a list of edges building the cell."""
+        return [self.get_edge(i) for i in range(self.n_edges)]
+
+    def get_edge(self, i) -> Cell:
+        """Get the i-th edge building the cell."""
+        return Cell(self.GetEdge(i))
+
+    @property
+    def faces(self) -> List[Cell]:
+        """Get a list of cell faces."""
+        return [self.get_face(i) for i in range(self.n_faces)]
+
+    def get_face(self, i) -> Cell:
+        """Get the i-th face building the cell."""
+        return Cell(self.GetFace(i))
+
+    @property
+    def bounds(self) -> Tuple[float, float, float, float, float, float]:
+        """Get the cell bounds [xmin, xmax, ymin, ymax, zmin, zmax]."""
+        return self.GetBounds()
+
+    def _get_attrs(self):
+        """Return the representation methods (internal helper)."""
+        attrs = []
+        attrs.append(("Type", str(self.type), "{}" * len(str(self.type))))
+        attrs.append(("Linear", self.is_linear, "{}"))
+        attrs.append(("Dimension", self.dimension, "{}"))
+        attrs.append(("N Points", self.n_points, "{}"))
+        attrs.append(("N Faces", self.n_faces, "{}"))
+        attrs.append(("N Edges", self.n_edges, "{}"))
+        bds = self.bounds
+        fmt = f"{pyvista.FLOAT_FORMAT}, {pyvista.FLOAT_FORMAT}"
+        attrs.append(("X Bounds", (bds[0], bds[1]), fmt))
+        attrs.append(("Y Bounds", (bds[2], bds[3]), fmt))
+        attrs.append(("Z Bounds", (bds[4], bds[5]), fmt))
+
+        return attrs
+
+    def str(self, html=False):
+        """Return a string with the cell info."""
+        fmt = ""
+        if html:
+            return "Not implemented yet"
+
+        # Otherwise return a string that is Python console friendly
+        fmt = f"{type(self).__name__} ({hex(id(self))})\n"
+        # now make a call on the object to get its attributes as a list of len 2 tuples
+        row = "  {}:\t{}\n"
+        for attr in self._get_attrs():
+            try:
+                fmt += row.format(attr[0], attr[2].format(*attr[1]))
+            except:
+                fmt += row.format(attr[0], attr[2].format(attr[1]))
+        return fmt
+
+    def __repr__(self):
+        """Return the object representation."""
+        return self.str(html=False)
+
+    def __str__(self):
```

Comment:
I think we don't need this as it will default to using `__repr__`, which is currently the same.

Reply:
I agree